apps/server/lib/lexical/server/boot.ex

```diff
@@ -16,16 +16,31 @@ defmodule Lexical.Server.Boot do
 
   def start do
     {:ok, _} = Application.ensure_all_started(:mix)
+
     Application.stop(:logger)
     load_config()
     Application.ensure_all_started(:logger)
 
     Enum.each(@dep_apps, &load_app_modules/1)
-    verify_packaging()
-    verify_versioning()
+
+    case detect_errors() do
+      [] ->
+        :ok
+
+      errors ->
+        errors
+        |> Enum.join("\n\n")
+        |> halt()
+    end
+
     Application.ensure_all_started(:server)
   end
 
+  @doc false
+  def detect_errors do
+    List.wrap(packaging_errors()) ++ List.wrap(versioning_errors())
```

Comment:
versioning_errors is always a list, i think that the naming indicates that `packaging_errors` should also be a list, but it can be nil. I think you should move `List.wrap` to `packaging_errors` so this is just `Enum.concat(packaging_errors(), versioning_errors())`